src/d3a/models/strategy/area_agents/two_sided_pay_as_bid_engine.py

```diff
@@ -153,34 +140,59 @@ def event_bid_deleted(self, *, bid):
             return
 
         if bid_info.source_bid.id == bid_id:
-            # Bid in source market of an bid we're already offering in the target market
+            # Bid in source market of an bid we're already bidding the target market
             # was deleted - also delete in target market
             try:
                 self.delete_forwarded_bids(bid_info)
             except MarketException:
-                self.owner.log.exception("Error deleting InterAreaAgent offer")
+                self.owner.log.exception("Error deleting InterAreaAgent bid")
         self._delete_forwarded_bid_entries(bid_info.source_bid)
 
-    def event_bid_changed(self, *, market_id, existing_bid, new_bid):
+    def event_bid_split(self, *, market_id, original_bid, accepted_bid, residual_bid):
         market = self.owner._get_market_from_market_id(market_id)
         if market is None:
             return
 
-        if market == self.markets.target:
-            # one of our forwarded bids was split, so save the residual bid for handling
-            # the upcoming trade event
-            assert existing_bid.id not in self.bid_trade_residual, \
-                "Bid should only change once before each trade, there has been already " \
-                "an event_bid_changed for the same bid ({existing_bid.id})."
-            self.bid_trade_residual[existing_bid.id] = new_bid
-        if market == self.markets.source and existing_bid.id in self.forwarded_bids:
-            # a bid in the source market was split - forward the new residual bid
-            if not self.owner.usable_bid(existing_bid) or \
-                    self.owner.name == existing_bid.seller:
+        if market == self.markets.target and accepted_bid.id in self.forwarded_bids:
+            # bid was split in target market, also split in source market
+
+            local_bid = self.forwarded_bids[original_bid.id].source_bid
+            original_bid_price = local_bid.original_bid_price \
+                if local_bid.original_bid_price is not None else local_bid.price
+
+            local_split_bid, local_residual_bid = \
+                self.markets.source.split_bid(local_bid, accepted_bid.energy, original_bid_price)
+
+            #  add the new bids to forwarded_bids
+            self._add_to_forward_bids(local_residual_bid, residual_bid)
+            self._add_to_forward_bids(local_split_bid, accepted_bid)
+
+        elif market == self.markets.source and accepted_bid.id in self.forwarded_bids:
+            # bid in the source market was split, also split in the target market
```

Comment:
Better to add one more comment here, that this is the case that handles forwarded bids to markets other than the one that the trade will take place.